libcrux-ml-dsa/src/encoding/signing_key.rs

```diff
@@ -10,32 +10,23 @@ use crate::{
     simd::traits::Operations,
 };
 
-#[allow(non_snake_case)]
 #[inline(always)]
-pub(crate) fn generate_serialized<
-    SIMDUnit: Operations,
-    Shake256: shake256::DsaXof,
-    const ROWS_IN_A: usize,
-    const COLUMNS_IN_A: usize,
-    const ETA: usize,
-    const ERROR_RING_ELEMENT_SIZE: usize,
-    const SIGNING_KEY_SIZE: usize,
->(
-    seed_for_A: &[u8],
-    seed_for_signing: &[u8],
+pub(crate) fn generate_serialized<SIMDUnit: Operations, Shake256: shake256::DsaXof>(
+    eta: Eta,
+    error_ring_element_size: usize,
+    seed_matrix: &[u8],
+    seed_signing: &[u8],
     verification_key: &[u8],
-    s1: [PolynomialRingElement<SIMDUnit>; COLUMNS_IN_A],
-    s2: [PolynomialRingElement<SIMDUnit>; ROWS_IN_A],
-    t0: [PolynomialRingElement<SIMDUnit>; ROWS_IN_A],
-) -> [u8; SIGNING_KEY_SIZE] {
-    let mut signing_key_serialized = [0u8; SIGNING_KEY_SIZE];
+    s1_2: &[PolynomialRingElement<SIMDUnit>],
```

Comment:
does this argument name come from the ML-DSA standard? if so, its ok, otherwise, a more informative name would be useful.